### ISSUE TITLE
Group context extension checks

### ISSUE THREAD

**Issue by franziskuskiefer** [openmls/openmls#1589]
Test all checks for groups context extension proposals.

**Comment by keks** [comment]
We recently added a test for a check that we hadn't tested before: https://github.com/openmls/openmls/pull/1590

I'll go through the list of required checks today and see whether we now have everything.

**Comment by keks** [comment]
Checks:
- [ ] [103](https://openmls.tech/validation.dhall/#103): Verify that the LeafNode is compatible with the group's parameters. If the GroupContext has a required_capabilities extension, then the required extensions, proposals, and credential types MUST be listed in the LeafNode's capabilities field. 
  - (when adding/updating a leaf node)
- [ ] [201](https://openmls.tech/validation.dhall/#201): Verify that the cipher suite and protocol version of the KeyPackage match those in the GroupContext. 
- [x] [308](https://openmls.tech/validation.dhall/#308):  For a regular, i.e., not external, Commit, the list is invalid if any of the following occurs: [...] It contains multiple GroupContextExtension proposals.
  - [validation test on current main](https://github.com/openmls/openmls/blob/2f835f789d5ef74f812aca37a823eaf7245ffb55/openmls/src/group/mls_group/test_mls_group.rs#L1297-L1303)
- [ ] [1001](https://openmls.tech/validation.dhall/#1001): A GroupContextExtensions proposal is invalid if it includes a required_capabilities extension and some members of the group do not support some of the required capabilities (including those added in the same Commit, and excluding those removed). 
  - [commit creation test on current main](https://github.com/openmls/openmls/blob/2f835f789d5ef74f812aca37a823eaf7245ffb55/openmls/src/group/mls_group/test_mls_group.rs#L1343-L1353)
- [ ] [1002](https://openmls.tech/validation.dhall/#1002): For new members, support for required capabilities is enforced by existing members during the application of Add commits. Existing members should of course be in compliance already. In order to ensure this continues to be the case even as the group's extensions are updated, a GroupContextExtensions proposal is deemed invalid if it contains a required_capabilities extension that requires non-default capabilities not supported by all current members. 


So far only one of these is properly tested. However, that test can serve as a blueprint for building tests for the others.Ideally each should get its individual test, though.

**Comment by keks** [comment]
I restructured the test code for group context extensions a little bit. I wasn't sure how to best use the test framework, so what I did is I moved some common state and functionality used in the GCE tests into types of a GCE test module. The tests are also in there.

I also added a test that checks that we don't accept commits that add members with insufficient capabilities. What is still missing to completely cover 103 is that this also prevents updates, I think (correct me if I'm wrong). I updated the listing above,
